src/services/mentees.js

```diff
@@ -925,8 +925,42 @@ module.exports = class MenteesHelper {
 						relatedOrgs = userOrgDetails.data.result.related_orgs
 					}
 					if (visibilityPolicy === common.ASSOCIATED) {
-						organization_ids.push(...relatedOrgs)
+						const associatedAdditionalFilter =
+							filter_type.toLowerCase() == common.MENTEE_ROLE
+								? {
+										external_session_visibility_policy: {
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+								: {
+										mentor_visibility_policy: {
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+
+						const organizationExtension = await organisationExtensionQueries.findAll(
+							{
+								[Op.and]: [
+									{
+										organization_id: {
+											[Op.in]: [...relatedOrgs, orgPolicies.organization_id],
```

Comment:
@adithyadinesh0412 remove this from here orgPolicies.organization_id

add  this to  organization_ids []  by default